apps/api/src/employee/employee.controller.ts

```diff
@@ -28,9 +30,15 @@ export class EmployeeController {
   ) {}
 
   @UseGuards(AuthGuard)
-  @Get('/:userid')
-  async getEmployee(@Req() req: RequestObj): Promise<EmployeeEntity> {
-    return req.user;
+  @Get()
+  async getEmployee(@Req() req: RequestObj, @Query('id') id: string): Promise<EmployeeRO> {
+    if (!id) return req.user;
+
+    const employee = await this.employeeService.getEmployee(id);
+    if (!employee) {
+      throw new NotFoundException(`User with id '${id}' not found`);
+    }
```

Comment:
This route may need a bit of a refactor. The intention behind it was to have a route to get the user object of the logged in user. All the work is done by the guard and it should be returned. 

It looks like you're adding a new purpose to the route, eg getting another user's user object. We might want to split this into another route. We should also be careful about what information we send back to a user who is not the logged in user.

Reply:
Agree with David's point here.
We can easily control with auth guard, only the authorized(super user) user to reach this route controller.